cluster-autoscaler/cloudprovider/gke/autoscaling_gke_client_v1beta1.go

```diff
@@ -93,22 +91,16 @@ func (m *autoscalingGkeClientV1beta1) FetchNodePools() ([]NodePool, error) {
 			})
 		}
 	}
-	return nodePools, nil
-}
-
-func (m *autoscalingGkeClientV1beta1) FetchLocations() ([]string, error) {
-	cluster, err := m.gkeBetaService.Projects.Locations.Clusters.Get(m.clusterPath).Do()
-	return cluster.Locations, err
+	return Cluster{
+		Locations:       clusterResponse.Locations,
+		NodePools:       nodePools,
+		ResourceLimiter: buildResourceLimiter(clusterResponse),
+	}, nil
 }
 
-func (m *autoscalingGkeClientV1beta1) FetchResourceLimits() (*cloudprovider.ResourceLimiter, error) {
-	cluster, err := m.gkeBetaService.Projects.Locations.Clusters.Get(m.clusterPath).Do()
-	if err != nil {
-		return nil, err
-	}
+func buildResourceLimiter(cluster *gke_api_beta.Cluster) *cloudprovider.ResourceLimiter {
 	if cluster.Autoscaling == nil {
-		glog.Warningf("FetchResourceLimits called without autoscaling limits set")
-		return nil, nil
+		return nil
```

Comment:
I don't really expect to get here, so maybe we should keep the warning?

Reply:
Done